### PR TITLE
Fixed bug where Quick Query would buffer the entire query result befo…

### PR DESCRIPTION
**Pull request by seanmcc-msft**
…re parsing it

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 1]
```diff
@@ -1971,7 +1971,6 @@ private void SetNameFieldsIfNull()
                         sourceIfUnmodifiedSince: sourceConditions?.IfUnmodifiedSince,
                         sourceIfMatch: sourceConditions?.IfMatch,
                         sourceIfNoneMatch: sourceConditions?.IfNoneMatch,
-                        sourceIfTags: sourceConditions?.TagConditions,
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1974]: this doesn't seem related.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1974]: It’s not, that is to pass CI. I think we need to revert the swagger change that caused this.

**Comment by seanmcc-msft** [comment]
@pakrym, could you take a look at this?

File: sdk/storage/Azure.Storage.Internal.Avro/src/AvroReader.cs [hunk 2]
```diff
@@ -60,14 +60,28 @@ internal class AvroReader
...
-            _dataStream = dataStream;
-            _headerStream = dataStream;
+            if (dataStream.CanSeek)
+            {
+                _dataStream = dataStream;
+                _headerStream = dataStream;
+            }
+            else
+            {
+                _dataStream = new StreamWithPosition(dataStream);
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Internal.Avro/src/AvroReader.cs:81]: Can you track position as part of `AvroReader` itself?

**Comment by pakrym** [comment]
LGTM

File: sdk/storage/Azure.Storage.Internal.Avro/src/AvroReader.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Internal.Avro/src/AvroReader.cs:81]: I could, but it would be extremely non-trivial.  Adding `StreamWithPosition` was much less disruptive, and seemed like the safer option.

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: Resolved this issue.

**Comment by seanmcc-msft** [comment]
@kasobol-msft, @amnguye, @jaschrep-msft

**Review by kasobol-msft — APPROVED** [review]
